control-plane/connect-inject/webhook/redirect_traffic.go

```diff
@@ -19,16 +19,35 @@ import (
 // iptables.Config:
 //
 //	ConsulDNSIP: an environment variable named RESOURCE_PREFIX_DNS_SERVICE_HOST where RESOURCE_PREFIX is the consul.fullname in helm.
-//	ProxyUserID: a constant set in Annotations
+//	ProxyUserID: a constant set in Annotations or read from namespace when using OpenShift
 //	ProxyInboundPort: the service port or bind port
 //	ProxyOutboundPort: default transparent proxy outbound port or transparent proxy outbound listener port
 //	ExcludeInboundPorts: prometheus, envoy stats, expose paths, checks and excluded pod annotations
 //	ExcludeOutboundPorts: pod annotations
 //	ExcludeOutboundCIDRs: pod annotations
 //	ExcludeUIDs: pod annotations
 func (w *MeshWebhook) iptablesConfigJSON(pod corev1.Pod, ns corev1.Namespace) (string, error) {
-	cfg := iptables.Config{
-		ProxyUserID: strconv.Itoa(sidecarUserAndGroupID),
+	cfg := iptables.Config{}
+
+	if !w.EnableOpenShift {
```

Comment:
Should this be `if w.EnableOpenShift {..}`?
If I'm reading the code correctly, this block of code is attempting to get uids from ns to exclude from traffic redirection which is specific to OpenShift?

Reply:
The else statement handles getting the uids from the namespace, not this block.